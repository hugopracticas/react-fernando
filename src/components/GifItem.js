import React from "react";
import PropTypes from "prop-types";

const GifItem = ({ title, url, id }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p data-testid="test-title">{title}</p>
    </div>
  );
};

export default GifItem;

GifItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
};
