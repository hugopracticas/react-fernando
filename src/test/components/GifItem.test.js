import { getByTestId, render, screen } from "@testing-library/react";
import GifItem from "../../components/GifItem";

describe("Test en <GifItem />", () => {
  const title = "Avatar";
  const url = "https://avatar.com/avatar.jpg";

  test("Should match with the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("debe de mostrar la imagen con la URL y el ALT indicado", () => {
    render(<GifItem title={title} url={url} />);
    //screen.debug();
    //expect(screen.getByRole("img").src).toBe(url);
    //expect(screen.getByRole("img").alt).toBe(title);
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(alt);
  });

  test("debe de mostrar el titulo", () => {
    const { getByTestId } = render(<GifItem title={title} url={url} />);
    //screen.debug();
    expect(screen.getByText(title)).toBeTruthy();
    expect(getByTestId("test-title").innerHTML).toBe(title);
  });
});
