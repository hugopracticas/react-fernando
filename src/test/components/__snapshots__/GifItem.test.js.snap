// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en <GifItem /> Should match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Avatar"
      src="https://avatar.com/avatar.jpg"
    />
    <p
      data-testid="test-title"
    >
      Avatar
    </p>
  </div>
</div>
`;
